### ISSUE TITLE
Healthcheck on SSL fails if the SSL certificate is invalid

### ISSUE THREAD

**Issue by Gui13** [palantir/docker-compose-rule#199]
If my healthcheck consists of an SSL endpoint, the test rule will fail if the certificate is invalid.

Although I can understand the point when the test is made for pre production purposes, it means that internal testing (or local testing) will fail if the healthcheck needs to reach an SSL endpoint that has no valid SSL cert.

Would it be possible to add a flag somewhere to allow for invalid SSL certificate in health checks?

**Comment by jesper-beck-holm** [comment]
Hi,
I have same experience with warning:
DockerExecutionException: 'docker inspect --format={{if not .State.Running}}DOWN{{else if .State.Paused}}PAUSED{{else if index .State "Health"}}{{if eq .State.Health.Status "healthy"}}HEALTHY{{else}}UNHEALTHY{{end}}{{else}}HEALTHY{{end}} /tmp/_MEIaSkdza/requests/packages/urllib3/connection.py:251: SecurityWarning: Certificate has no `subjectAltName`, falling back to check for a `commonName` for now. This feature is being removed by major browsers and deprecated by RFC 2818. (See https://github.com/shazow/urllib3/issues/497 for details.)
    19c3c711eab74754257daa1153aa9d6c0b30618fe204be7aeb247a9cd60ce840' **returned exit code 1**
    The output was:

    Error: No such object: /tmp/_MEIaSkdza/requests/packages/urllib3/connection.py:251: **SecurityWarning**: Certificate has no `subjectAltName`, falling back to check for a `commonName` for now. This feature is being removed by major browsers and deprecated by RFC 2818. (See https://github.com/shazow/urllib3/issues/497 for details.)

By the way, Great job!

**Comment by pkoenig10** [comment]
You can avoid SSL errors by creating an always trusting `SSLSocketFactory` and setting it as the default:
```
static {
    HttpsURLConnection.setDefaultSSLSocketFactory(sslSocketFactory);
}
```

**Comment by Gui13** [comment]
Yes, this is a solution, but this means you're sacrificing all security on all HttpsUrlConnections, not just the ones that DockerComposeRule uses.

**Comment by pkoenig10** [comment]
Well you don't have to use an always-trusting one, you can provide a custom truststore.
